source/fem-dem/ib_particles_dem.cc

```diff
@@ -0,0 +1,818 @@
+//
+// Created by lucka on 2021-10-11.
+//
+#include <fem-dem/ib_particles_dem.h>
+
+
+template <int dim>
+void
+IBParticlesDEM<dim>::initialize(SimulationParameters<dim> p_nsparam,
+                                MPI_Comm &mpi_communicator_input,
+                                std::vector<IBParticle<dim>> particles)
+{
+  parameters       = p_nsparam;
+  mpi_communicator = mpi_communicator_input;
+  dem_particles    = particles;
+  boundary_cells.resize(dem_particles.size());
+}
+template <int dim>
+void
+IBParticlesDEM<dim>::update_particles(std::vector<IBParticle<dim>> particles,
+                                      double                       time)
+{
+  dem_particles = particles;
+  cfd_time      = time;
+}
+
+
+template <int dim>
+void
+IBParticlesDEM<dim>::calculate_pp_contact_force(
```

Comment:
I think I can use the function in the DEM class instead of rewriting the contact force calculations. this is very dangerous because when I modify the DEM contact model or correct a bug there, it's not updated here. For instance, there was a mistake in the calculation of the tangential torque that I corrected in the DEM solver. but it's still here. The tangential torque acting on particles i and j are different and it should be calculated using their specific radii, not the effective radius. So I will see tomorrow if I can use the contact force class of the DEM here and solve all these issues and possible future problems.

Reply:
Good idea @shahabgol 
I'll wait for this to review. We just need to prevent code duplication as much as possible. We did a pretty good job with the unresolved CFD-DEM for that.